app/src/app/views/stix/collection/collection-import/collection-import-workflow/collection-import.component.ts

```diff
@@ -249,6 +257,56 @@ export class CollectionImportComponent implements OnInit {
         })
     }
 
+    /**
+     * Determine if the user can force import a collection after the post collection
+     * call fails. Users may force an import when:
+     *   1. the collection bundle is already in the database (user is re-importing the bundle)
+     *   2. the collection bundle has one or more objects with an invalid ATT&CK spec version.
+     * @param error the resulting error from previewing the collection
+     * @returns true if the user can force import the collection; false otherwise
+     */
+    private canForceImport(err: any): boolean {
+        if (!err.error) return false;
+        if (err.error.bundleErrors.duplicateCollection ||
+            err.error.objectErrors.summary.invalidAttackSpecVersionCount) return true;
+        return false;
+    }
+
+    /**
+     * Set up the objects to force import from a collection
+     * @param collectionBundle the collection bundle to import
+     * @param error the resulting error from previewing the collection
+     */
+    public errorPreview(collectionBundle: any, error: any): void {
+        this.import_error = error;
+        this.collectionBundle = collectionBundle; //save for later
+
+        let invalidObjects: string[] = error.objectErrors.errors.map(obj => obj.id);
+        let selected: string[] = [];
+        for (let object of this.collectionBundle.objects) {
+            if ("type" in object && object.type == 'x-mitre-collection') continue;
+            if ("id" in object && !invalidObjects.includes(object.id)) {
+                selected.push(object.id);
+            }
+        }
+        this.select =  new SelectionModel(true, selected);
+        this.stepper.next();
+    }
+
+    /**
+     * Handle user selection on collection import errors
+     * @param force if true, force import the collection
+     */
+    forceImport(force: boolean) {
+        if (force) {
+            // proceed to force import the collection
+            this.import(true);
+        } else {
+            // user cancelled, revert to previous step
+            this.stepper.previous();
+        }
+    }
```

Comment:
This function is confusingly named. It's "forceImport", but depending on the argument it could also be "cancelImport". Might want to rename it.